tests/utils/test_db_cleanup.py

```diff
@@ -207,7 +207,7 @@ def test__build_query(self, table_name, date_add_kwargs, expected_to_delete, ext
         target_table_name = "_airflow_temp_table_name"
         with create_session() as session:
             clean_before_date = base_date.add(**date_add_kwargs)
-            query = _build_query(
+            query, delete_stmt = _build_query(
```

Comment:
i think it would be nice to add asserts about the queries so we can see how they changed

e.g. 
```
            assert (
                str(query)
                == """SELECT base.*
FROM task_instance AS base
WHERE base.start_date < ?"""
            )
            assert (
                str(delete_stmt) == "DELETE FROM task_instance AS base WHERE base.start_date < :start_date_1"
            )
```

Reply:
in fact would be really nice to do it in two commits.  first first commit adding these asserts in the tests and then in the second commit that would let us easily verify that the only changes are desirable. but otherwise everything looks good and i tried locally and seems to be right.

Reply:
thank you for the feedback! Do you think it would be a good idea to add the SQL for dag_run as well, as it has a different logic for delete? SQLs are mentioned in the above comment

Reply:
Yeah I feel it would be good for any query we are testing.  It’s pretty easy to add an asset for the compiled sql and it is great for review